app/controllers/mixins/middleware_common_mixin.rb

```diff
@@ -25,4 +29,35 @@ def clear_topology_breadcrumb
       @breadcrumbs.clear
     end
   end
+
+  def init_show(model_class = self.class.model)
+    clear_topology_breadcrumb
+    @lastaction = 'show'
+    @showtype = 'main'
+    @gtl_url = '/show'
+    @display = params[:display] || 'main' unless control_selected?
```

Comment:
this was not the original intent of the PR, the purpose was to break the dependency in code from containers provider and to increase code reuse. I'm not against improving more, but don't think it's necessarily should be done here. I'll leave the decision to @rubenvp8510 whether to address it here or in a following PR